x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -507,6 +455,79 @@ private void checkTagsAgainstModelIds(List<String> tags, ActionListener<Void> li
         );
     }
 
+    private boolean validateModelDefinition(TrainedModelConfig config, ClusterState state, ActionListener<Response> finalResponseListener) {
```

Comment:
There is a yaml LTR test which is passing because it runs with the trial licence, which is good. It's a shame the same test can't cover the negative case where the licence is not sufficient. 

https://github.com/elastic/elasticsearch/blob/248b045d7029c548d6c5aa149b728c314582c88b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/learning_to_rank_rescorer.yml

Is it possible to make this function static and add a tests to `TransportPutTrainedModelActionTests` to cover the cases where there is a basic licence and the validation fails.

Reply:
> Is it possible to make this function static and add a tests to TransportPutTrainedModelActionTests to cover the cases where there is a basic licence and the validation fails.

I'll take a look. Thanks!

Reply:
I've updated this a bit, so I re-requested another set of 👀 on this before I'm confident to merge...